tendrl/gluster_integration/sds_sync/utilization.py

```diff
@@ -0,0 +1,105 @@
+import etcd
+
+
+def sync_utilization_details(volumes):
+    cluster_raw_capacity = 0
+    cluster_used_capacity = 0
+    cluster_usable_capacity = 0
+    for volume in volumes:
+        subvol_count = 0
+        bricks = []
+        raw_capacity = 0
+        raw_used = 0
+        up_bricks = 0
+        subvols = []
+        while True:
+            try:
+                subvol = NS._int.client.read(
+                    "clusters/%s/Volumes/%s/Bricks/subvolume%s" % (
+                        NS.tendrl_context.integration_id,
+                        volume.vol_id,
+                        subvol_count
+                    )
+                )
+                subvol_bricks = []
+                for entry in subvol.leaves:
+                    brick_name = entry.key.split("/")[-1]
+                    fetched_brick = NS.gluster.objects.Brick(
+                        name=brick_name
+                    ).load()
+                    raw_capacity += fetched_brick.utilization['total']
+                    cluster_raw_capacity += \
+                        fetched_brick.utilization['total']
+                    raw_used += fetched_brick.utilization['used']
+                    cluster_used_capacity += \
+                        fetched_brick.utilization['used']
+                    bricks.append(fetched_brick)
+                    subvol_bricks.append(fetched_brick)
+                    if fetched_brick.status != "Started":
+                        up_bricks += 1
+                subvols.append(subvol_bricks)
+                subvol_count += 1
+            except etcd.EtcdKeyNotFound:
+                break
+        vol_usable_capacity = 0
+        vol_used_capacity = 0
+        vol_pcnt_used = 0
+        if up_bricks == len(bricks):
+            if int(volume.disperse_count) == 0:
+                # This is distribute or replicate volume
+                vol_usable_capacity = \
```

Comment:
This logic is copied from `gstatus` as is and based on disperse_yield value only data part is used for calculation of utilization details